### PR TITLE
groups: attempt to handle equivalent e-mail addresses

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Part of: https://github.com/kubernetes/k8s.io/issues/2890

Attempt to deal with email addresses that are basically equivalent by 
comparing email address with case insensitivity, and stripping dots from 
the local part (before the '@').

These rules make sense for gmail addresses, which comprise the majority of
email addresses used in our groups. It's difficult to know whether they
apply universally to all e-mail addresses.

https://www.rfc-editor.org/rfc/rfc5321#section-2.3.11 says that the 
semantics for the local part of an e-mail address are determined by the 
host. So the only way to know for sure if two email addresses are 
equivalent is to send e-mail to both of them, and inspect the resulting 
mailbox.

It sure seems like there must be some part of the Directory API that knows
how to handle this equivalence, so I wanted to rely on that instead of
trying to build our own hackish local best guess at email equivalence.  But
when I tried using either the GetMember or HasMember API calls with the
semantically equivalent email addresses, I got back a 404 not found, and
400 invalid member key respectively.

So, we'll go with this for now until we receive reports of someone not 
receiving email due to case-sensitivty or dot-sensitivity issues.

Other things included in here:

- To aid in debugging I hacked up my group config to focus just on the
sig-release groups, which led me to add support for relative paths in the
config file.

- I had initially hoped refreshing libraries might get us some kind of fix
for free, so `go.mod` and `go.sum` have a few updates (see that commit
for details)

### CODE REVIEW

**Comment by spiffxp** [comment]
If you look at the linked issue there are two other domains failing due to case insensitivity

**Comment by MadhavJivrajani** [comment]
Ah, missed that. Thanks for clarifying :)
LGTM!

**Comment by spiffxp** [comment]
ping @nikhita PTAL

**Comment by spiffxp** [comment]
Rebased to avoid merge conflict

**Comment by spiffxp** [comment]
Missed merge conflict in boilerplate, fixed

File: groups/service.go [hunk 1]
```diff
@@ -503,3 +520,26 @@ func deepCopySettings(a, b interface{}) {
 	byt, _ := json.Marshal(a)
 	json.Unmarshal(byt, b)
 }
+
+// EmailAddressEquals checks equivalence between two e-mail addresses according
+// to the following rules:
+// - email addresses are case-insensitive (e.g. FOO@bar.com == foo@bar.com)
+// - local parts are dot-inensitive (e.g. foo@bar.com == f.o.o@bar.com)
+func EmailAddressEquals(a, b string) bool {
```
Comment (nikhita) [inline groups/service.go:528]: Should we be considering the "anything after `+` is ignored" case here?

**Review by nikhita — APPROVED** [review]
Apologies for the delay. One question but we can add that as a follow-up if needed and this lgtm so:

/lgtm

**Pull request merged**